src/common/translation.cpp

```diff
@@ -1887,12 +1887,28 @@ wxMsgCatalog *wxResourceTranslationsLoader::LoadCatalog(const wxString& domain,
     size_t mo_size = 0;
 
     const wxString resname = wxString::Format("%s_%s", domain, lang);
+    // Replace non-alphanumeric characters with `_`
+    wxString resname_safe = resname;
+    for ( wxString::iterator it = resname_safe.begin(); it != resname_safe.end(); ++it )
+    {
+        if ( !( (*it >= (wxChar) 'A' && *it <= (wxChar) 'Z') ||
+                (*it >= (wxChar) 'a' && *it <= (wxChar) 'z') ||
+                (*it >= (wxChar) '0' && *it <= (wxChar) '9') ||
+                *it == (wxChar) '_') )
+        {
+                *it = (wxChar) '_';
+        }
+    }
 
     if ( !wxLoadUserResource(&mo_data, &mo_size,
                              resname,
                              GetResourceType().t_str(),
                              GetModule()) )
-        return NULL;
+        if ( !wxLoadUserResource(&mo_data, &mo_size,
+                                resname_safe,
+                                GetResourceType().t_str(),
+                                GetModule()) )
+            return NULL;
```

Comment:
I don't really think we need to try both, if using non-alnum characters doesn't work in any case, using just the "safe" name should be enough.

But if you really want to do it like this, then it would make sense to move the construction of the safe name here, as otherwise we're not going to use it at all in 99% of cases.

Reply:
The original 'if' code was untouched because I wanted to preserve some backward-compatibility. 

The problem is little complicated, as is described [here](https://github.com/wxWidgets/wxWidgets/issues/22589#issuecomment-1173975641) (This kind of resource id is valid in MSVC by adding quotes around it but not valid in MinGW windres. The naming rule of resource id is undocumented so it varies on different implementations). If some one has already using `"myapp_sr@latin"` in MSVC, their app will not load it in the future version of wxWidgets.

` | `myapp_sr@latin` | `"myapp_sr@latin"` | `myapp_sr_latin`
-----|-----|-----|-----
MSVC rc.exe | ❎ (Syntax error) | ✅ | ✅
GNU windres | ❎ (Syntax error) | ❎ (Buggy, unrelated error output)　| ✅
llvm-rc | ❎ (Syntax error) | ❎ (Syntax error)　| ✅


Yes I think 99% people won't met this case, so I finally decided to remove this check to make the code clean. Now both MSVC and MinGW should follow the last rule (`myapp_sr_latin`).